lua/weapons/weapon_peacekeeper.lua

```diff
@@ -0,0 +1,341 @@
+SWEP.Secondary.Delay 		= 0
+SWEP.Secondary.Damage 		= 0
+
+sound.Add({
+	name = 			"Double_Barrel.Single",
+	channel = 		CHAN_USER_BASE+10,
+	volume = 		1.0,
+	sound = 			"weapons/peacekeeper/peacekeeper_fire.wav"
+})
+
+sound.Add({
+	name = 			"Double_Barrel.InsertShell",
+	channel = 		CHAN_ITEM,
+	volume = 		1.0,
+	sound = 			"weapons/peacekeeper/xm1014_insertshell.mp3"
+})
+
+sound.Add({
+	name = 			"Double_Barrel.barreldown",
+	channel = 		CHAN_ITEM,
+	volume = 		1.0,
+	sound = 			"weapons/peacekeeper/barreldown.mp3"
+})
+
+sound.Add({
+	name = 			"Double_Barrel.barrelup",
+	channel = 		CHAN_ITEM,
+	volume = 		1.0,
+	sound = 			"weapons/peacekeeper/barrelup.mp3"
+})
+
+function SWEP:Deploy()
+	if not IsValid(self) then return end
+	if not IsValid(self.Owner) then return end
+	if not self.Owner:IsPlayer() then return end
+	
+	self:SetHoldType(self.HoldType)
+	
+	local timerName = "ShotgunReload_" ..  self.Owner:UniqueID()
+	if (timer.Exists(timerName)) then
+		timer.Destroy(timerName)
+	end
+
+	self.Weapon:SendWeaponAnim(ACT_VM_DRAW)
+
+	self.Weapon:SetNextPrimaryFire(CurTime() + .25)
+	self.Weapon:SetNextSecondaryFire(CurTime() + .25)
+	self.ActionDelay = (CurTime() + .25)
+	
+	self.Owner.NextReload = CurTime() + 1
+
+	return true
+end
+
+function SWEP:SprintMod()
+	return ((math.cos(((self.SprintNess or 0)+1)*math.pi)+1)*0.5)
+end
+
+function SWEP:IsSprinting()
+	return self.Owner:GetVelocity():Length() > (self.Owner:GetWalkSpeed()+self.Owner:GetRunSpeed())*0.5
+end
+
+function SWEP:DrawWorldModel()
+	self:DrawModel()
+end
+
+function SWEP:GetViewModelPosition(pos,ang)
+	pos= pos + (0*ang:Up()) + (1*ang:Right()) + (0*ang:Forward())
+	ang:RotateAroundAxis(ang:Right(),-2)
+
+	local sm = self:SprintMod()
+
+	pos= pos + (0*sm*ang:Up()) + (1*sm*ang:Right())
+	ang:RotateAroundAxis(ang:Up(),sm*20)
+	ang:RotateAroundAxis(ang:Right(),sm*-20)
+
+	return pos,ang
+end
+
+function SWEP:DrawHUD()
+	local mx = ScrW()/2
+	local my = ScrH()/2
+
+	self.XHairspread = (EyePos()+EyeAngles():Forward()+EyeAngles():Right()*self:GetCone()):ToScreen().x - mx
+
+	local spread = self.XHairspread
+	local len = 10
+	local mx = ScrW()/2
+	local my = ScrH()/2
+
+	surface.SetDrawColor(255,200,20,Lerp(math.Clamp((self:GetCone()-0.3)*10,0,1),255,0))
+
+	surface.DrawLine(mx-(spread+len),my,mx-spread,my)
+	surface.DrawLine(mx+(spread+len),my,mx+spread,my)
+	surface.DrawLine(mx,my-(spread+len),mx,my-spread)
+	surface.DrawLine(mx,my+(spread+len),mx,my+spread)
+end
+
+if CLIENT then
+	function SWEP:Think()
+	
+		ply = self.Owner
+		if LocalPlayer():GetActiveWeapon()~=self then return end
+
+		local changesprintness = 2.2*RealFrameTime()
+	
+		if self:IsSprinting() then
+			changesprintness = -1*changesprintness
+		end
+
+		self.SprintNess = math.Clamp((self.SprintNess or 0) - changesprintness,0,1)
+
+	end
+end
+
+function SWEP:Initialize()
+	self:SetWeaponHoldType( self.HoldType )
+	self:SetHoldType( self.HoldType )
+end
+
+function SWEP:GetCone()
```

Comment:
Did you adjust the cone size from what the garand had? it should be big because its a shotgun. You can remove the sprint logic as well if you don't think its needed

Reply:
I did change the cone size